File: models/Team.rb
class Team
    attr_accessor :id, :name, :venue, :abbreviation, :firstYearOfPlay, :divisionId, :conferenceId, :franchiseId, :active

    def self.fromJson(item)
        newTeam = Team.new

        newTeam.id = item["id"]
        newTeam.name = item["name"]
        newTeam.venue = item["venue"]["name"]
        newTeam.abbreviation = item["abbreviation"]
        newTeam.firstYearOfPlay = item["firstYearOfPlay"]
        newTeam.divisionId = item["division"]["id"]
        newTeam.conferenceId = item["conference"]["id"]
        newTeam.franchiseId = item["franchise"]["franchiseId"]
        newTeam.active = item["active"]

        return newTeam
    end
end

File: domain/src/commonMain/kotlin/ca/sebleclerc/hockeydata/domain/Team.kt
package ca.sebleclerc.hockeydata.domain

data class Team(
  val id: Int,
  val name: String,
  val venue: String,
  val abbreviation: String,
  val firstYearOfPlay: String,
  val divisionId: Int,
  val conferenceId: Int,
  val franchiseId: Int,
  val active: Boolean,
) {
  companion object
}
